txpipe/psf_diagnostics.py

```diff
@@ -172,6 +172,415 @@ def plot_histogram(self, function, output_name, xlabel, edges):
         return results
 
 
+class TXTauStatistics(PipelineStage):
+    """
+    Compute and plot PSF Tau statistics where the definition of Tau stats are eq.20-22
+    of Gatti et al 2023.
+    """
+    name     = "TXTauStatistics"
+    parallel = False
+    inputs   = [("shear_catalog"           , ShearCatalog),
+                ("shear_tomography_catalog", TomographyCatalog),
+                ("star_catalog"            , HDFFile),
+                ("rowe_stats"              , HDFFile),
+               ]
+
+    outputs  = [
+                ("tau0p"    , PNGFile),
+                ("tau2p"    , PNGFile),
+                ("tau5p"    , PNGFile),
+                ("tau_stats", HDFFile),
+               ]
+
+
+    config_options = {
+                       "min_sep"       : 0.5,
+                       "max_sep"       : 250.0,
+                       "nbins"         : 20,
+                       "bin_slop"      : 0.01,
+                       "sep_units"     : "arcmin",
+                       "psf_size_units": "sigma",
+                     }
+
+    def run(self):
+        import treecorr
+        import h5py
+        import matplotlib
+
+        matplotlib.use("agg")
+
+        # Load galaxies
+        gal_ra, gal_dec, gal_g1, gal_g2, gal_weight   = self.load_galaxies()
+        gal_g = np.array((gal_g1, gal_g2))
+        
+        # Load star properties
+        ra, dec, e_psf, e_mod, de_psf, T_f, star_type = self.load_stars()
+
+        # Compute tau stats 
+        tau_stats  = {}
+        p_bestfits = {}
+
+        for t in STAR_TYPES:
+            if t==1:
```

Comment:
With this break the loop is redundant, as only t=0 will be used.

Reply:
Fixed.